boards/espressif/esp32c3_rust/doc/index.rst

```diff
@@ -0,0 +1,287 @@
+
++---------------------------+--------------+-----------------------------------------------------------------------------+--------------------------+---------+
+| Peripheral                | Part number  | Reference                                                                   | Crate                    | Address |
++===========================+==============+=============================================================================+==========================+=========+
+| IMU                       | ICM-42670-P  | `Datasheet <https://invensense.tdk.com/download-pdf/icm-42670-p-datasheet/>` | `Link <https://crates.io/crates/icm42670>`__   | 0x68    |
++---------------------------+--------------+-----------------------------------------------------------------------------+--------------------------+---------+
+| Temperature and Humidity  | SHTC3        | `Datasheet <https://www.sensirion.com/en/environmental-sensors/humidity-sensors/digital-humidity-sensor-shtc3-our-new-standard-for-consumer-electronics/>` | `Link <https://crates.io/crates/shtcx>`__       | 0x70    |
++---------------------------+--------------+-----------------------------------------------------------------------------+--------------------------+---------+
+
+I2C Bus Connection
+==================
+
++---------+--------+
+| Signal  | GPIO   |
++=========+========+
+| SDA     | GPIO10 |
++---------+--------+
+| SCL     | GPIO8  |
++---------+--------+
+
+I/Os
+====
+
+The following devices are connected through GPIO:
+
++--------------+--------+
+| I/O Devices  | GPIO   |
++==============+========+
+| WS2812 LED   | GPIO2  |
++--------------+--------+
+| LED          | GPIO7  |
++--------------+--------+
+| Button/Boot  | GPIO9  |
++--------------+--------+
+
+Power
+=====
+
+* USB type-C (*no PD compatibility*).
+* Li-Ion battery charger.
+
+System requirements
+*******************
+
+Prerequisites
+=============
+
+Espressif HAL requires WiFi and Bluetooth binary blobs in order work. Run the command
+below to retrieve those files.
+
+.. code-block:: console
+
+   west blobs fetch hal_espressif
+
+.. note::
+
+   It is recommended running the command above after :file:`west update`.
+
+Building & Flashing
+*******************
+
+Simple boot
+===========
+
+The board could be loaded using the single binary image, without 2nd stage bootloader.
+It is the default option when building the application without additional configuration.
+
+.. note::
+
+   Simple boot does not provide any security features nor OTA updates.
+
+MCUboot bootloader
+==================
+
+User may choose to use MCUboot bootloader instead. In that case the bootloader
+must be build (and flash) at least once.
+
+There are two options to be used when building an application:
+
+1. Sysbuild
+2. Manual build
+
+.. note::
+
+   User can select the MCUboot bootloader by adding the following line
+   to the board default configuration file.
+   ```
+   CONFIG_BOOTLOADER_MCUBOOT=y
+   ```
```

Comment:
see https://github.com/zephyrproject-rtos/zephyr/pull/75239

At some point (and apologies if I've already mentioned it) I think esp32 boards should all `.. include:: ...` common doc "snippets" rather than copy-paster over and over again. See e.g. https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/intel/adl/doc/index.rst#L62-L64

Reply:
I liked the idea, way cleaner. I will do that for all boards in another commit, ok?

Reply:
> see #75239
> 
> At some point (and apologies if I've already mentioned it) I think esp32 boards should all `.. include:: ...` common doc "snippets" rather than copy-paster over and over again. See e.g. https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/intel/adl/doc/index.rst#L62-L64

Oh, I totally agree with that.